plugin-test/src/test/groovy/org/gradle/github/dependency/extractor/MultiProjectDependencyExtractorTest.groovy

```diff
@@ -1,370 +1,242 @@
+            ],
+            "project :a"      : [
+                package_url : "pkg:maven/org.test/a@1.0",
+                relationship: "indirect",
+                dependencies: ["org.test:foo:1.0"]
+            ],
+            "org.test:foo:1.0": [
+                package_url : fooPurl,
+                relationship: "indirect"
+            ]
+
+        ])
     }
 
-    def "project with buildSrc"() {
+    def "extracts dependencies from buildSrc project"() {
         given:
-        multiProjectBuild("parent", []) {
-            project("buildSrc").tap {
-                buildFile """
-                apply plugin: 'java'
-                group = 'org.test.buildSrc'
-                version = '1.0'
-                dependencies {
-                    implementation 'org.test-published:foo:1.0'
-                }
-                """
-                setupBuildFile(it)
+        file("buildSrc/settings.gradle") << "rootProject.name = 'buildSrc'"
+        file("buildSrc/build.gradle") << """
+            apply plugin: 'java'
+            group = 'org.test.buildSrc'
+            version = '1.0'
+            repositories {
+                maven { url "${mavenRepo.uri}" }
+            }
+            dependencies {
+                implementation 'org.test:foo:1.0'
             }
+        """
 
-            buildFile """
+        buildFile << """
             apply plugin: 'java'
             dependencies {
-                implementation 'org.test-published:bar:2.0'
+                implementation 'org.test:bar:1.0'
             }
-            """
-            setupBuildFile(it)
-        }
+        """
+
         when:
         succeeds("validate")
 
         then:
-        Map manifests = jsonManifests()
-        def buildSrcClasspath = manifests["project :buildSrc"]
-
-        def buildSrcRuntimeFile = buildSrcClasspath.file as Map
-        buildSrcRuntimeFile.source_location == "buildSrc/build.gradle"
-        def buildSrcClasspathResolved = buildSrcClasspath.resolved as Map
-        def testFoo = buildSrcClasspathResolved[fooGav] as Map
-        verifyAll(testFoo) {
-            package_url == this.fooPurl
-            relationship == "direct"
-            dependencies == []
-        }
-        def runtimeClasspath = jsonManifest("project :")
-        def runtimeFile = runtimeClasspath.file as Map
-        runtimeFile.source_location == "build.gradle"
-        def runtimeClasspathResolved = runtimeClasspath.resolved as Map
-        def testBar = runtimeClasspathResolved[gavFor(bar)] as Map
-        verifyAll(testBar) {
-            package_url == purlFor(this.bar)
-            relationship == "direct"
-            dependencies == []
-        }
+        manifestNames == ["project :", "project :buildSrc"]
+
+        def buildSrcManifest = gitHubManifest("project :buildSrc")
+        buildSrcManifest.sourceFile == "buildSrc/build.gradle"
+        buildSrcManifest.assertResolved([
+            "org.test:foo:1.0": [package_url: fooPurl]
```

Comment:
In other tests the `package_url` is a literal and not a reference. Is there a reason the same isn't being done here? I actually really prefer the explicitness of the inline literal.

Reply:
The `package_url` value for an external module is long an unwieldy, and I think they are better _not_ inlined into the test. The `package_url` value for a project dependency is more succinct, and those have been inlined.

BUT, in other cases we generate the package_url using `purlFor(bar)` rather than a canned reference like `fooPurl`: this is likely more comprehensible. I'll make that change (and rename `purlFor()` to `packageUrlFor()`).